File: packages/capacitor/android/src/main/java/admob/plus/capacitor/ads/AdBase.java
package admob.plus.capacitor.ads;

import admob.plus.core.Ad;
import admob.plus.core.Helper;
import admob.plus.capacitor.ExecuteContext;

import static admob.plus.capacitor.ExecuteContext.plugin;

public class AdBase extends Ad {
    public AdBase(ExecuteContext ctx) {
        super(ctx);
    }

    @Override
    protected Helper.Adapter getAdapter() {
        return plugin;
    }
}

File: packages/capacitor/android/src/main/kotlin/admob/plus/capacitor/ads/AdBase.kt
package admob.plus.capacitor.ads

import admob.plus.capacitor.ExecuteContext
import admob.plus.core.Ad
import admob.plus.core.Helper

open class AdBase(ctx: ExecuteContext?) : Ad(ctx!!) {
    override val adapter: Helper.Adapter
        get() = ExecuteContext.Companion.plugin!!
}
